conn.go

```diff
@@ -344,6 +422,11 @@ func (c *Conn) Bind(sa unix.Sockaddr) error {
 // connected to a remote peer, Connect calls getpeername(2) and returns the
 // unix.Sockaddr from that call.
 func (c *Conn) Connect(sa unix.Sockaddr) (unix.Sockaddr, error) {
+	return c.ConnectContext(noContext, sa)
+}
+
+// ConnectContext is like Connect but takes a context.
+func (c *Conn) ConnectContext(ctx context.Context, sa unix.Sockaddr) (unix.Sockaddr, error) {
```

Comment:
As an example of what I mean, we can just make Connect:

```go
func (c *Conn) Connect(ctx context.Context, sa unix.Sockaddr) (unix.Sockaddr, error) {
```

And drop the suffix entirely.

Reply:
Haha, I am not sure if I have more knowledge of the BSD sockets, I wouldn't say I had any when I started this! Mostly a lot spelunking through the manual pages and net, os and internal/poll package.

I like to just put a context in as a first parameter, out of politeness and backward compatibility I opted to duplicate most methods. So remove the duplication. What about the comment about the order of parameters in the Sendto method? Adding a context parameter is probably a good moment to change the order as well as you have to change the call sides anyway.

Reply:
How do you want to go about Read and Write as *os.File provides a internal/poll implementation for them and if no context is required, I don't see any reason to use custom code. There will be an if though if a context is passed as the first parameter.

Reply:
> I like to just put a context in as a first parameter, out of politeness and backward compatibility I opted to duplicate most methods. So remove the duplication.

Yep I totally understand, but in this case I'd rather just break the few users of this package up front so we can fix it before a v1.0.0. I can tag a new minor version after we merge this.

> What about the comment about the order of parameters in the Sendto method? Adding a context parameter is probably a good moment to change the order as well as you have to change the call sides anyway.

Yes please, thanks. I forgot about that TODO.

> How do you want to go about Read and Write as *os.File provides a internal/poll implementation for them and if no context is required, I don't see any reason to use custom code. There will be an if though if a context is passed as the first parameter.

SGTM, if the sentinel noContext makes more sense then it seems fine to retain it in these cases. But I suspect for the majority of functions we could just pass a background context to keep it simple.